### PR TITLE
[Python]change extension of beam sdist from .zip to .tar.gz

### LINKED ISSUE #28629: [Task]: Modify release scripts to look for `.tag.gz` insead of `.zip` after introducing pyproject.toml
### What needs to happen?

with pyproject.toml, Beam is going to use `build` to build the sdists. The file format of the source is going to change from `.zip` to `.tar.gz`. 

Once https://github.com/apache/beam/pull/28385, create a follow up PR to make changes to the release scripts.

### Issue Priority

Priority: 3 (nice-to-have improvement)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes: https://github.com/apache/beam/issues/28629
https://github.com/apache/beam/pull/28385 changed the sdist extension from `zip` to `tar.gz`. Zip tarballs are not supported according to https://peps.python.org/pep-0517/#source-distributions

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/run_rc_validation.yml [hunk 1]
```diff
@@ -17,14 +17,14 @@
```
Comment (AnandInguva) [inline .github/workflows/run_rc_validation.yml:17]: This file has lots of whitespaces. My IDE automatically removes them.

**Comment by AnandInguva** [comment]
R: @damccorm @jrmccluskey

**Comment by jrmccluskey** [comment]
Please update documentation on https://cwiki.apache.org/confluence/display/BEAM/Python+Tips as well.

File: release/src/main/python-release/python_release_automation_utils.sh [hunk 2]
```diff
@@ -97,7 +97,7 @@ function download_files() {
 
     wget -r -l2 --no-parent -nd -A "$BEAM_PYTHON_SDK_WHL*" $RC_STAGING_URL
   else
-    BEAM_PYTHON_SDK_ZIP="apache-beam-$VERSION.zip"
+    BEAM_PYTHON_SDK_ZIP="apache-beam-$VERSION.tar.gz"
```
Comment (jrmccluskey) [inline release/src/main/python-release/python_release_automation_utils.sh:100]: This should be renamed

**Review by jrmccluskey — COMMENTED** [review]
How much validation have you done for these changes?

**Comment by AnandInguva** [comment]
> How much validation have you done for these changes?

I ran https://github.com/apache/beam/actions/runs/6616347730/job/17970424418  which publishes snapshot. It was not finding the target file before but with this change it works. 

I will find and run some more github actions to ensure they run correctly.

**Comment by AnandInguva** [comment]
> Please update documentation on https://cwiki.apache.org/confluence/display/BEAM/Python+Tips as well.

Done

**Comment by damccorm** [comment]
@jrmccluskey @AnandInguva what are next steps here? We should prioritize this one since its a release blocker and ideally would be in before wednesday

**Comment by jrmccluskey** [comment]
I think, based on the conversations we've had, this is fine to get in now.

**Review by jrmccluskey — APPROVED** [review]

**Pull request merged**